// react-lodash/src/Entries.js
import React from 'react'
import entries from 'lodash/entries'

export default function Entries(props) {
  return props.children(entries(props.object))
}

Entries.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.entries",
  "longname": "_.entries",
  "name": "entries",
  "kind": "member",
  "scope": "static",
  "description": "Creates an array of own enumerable string keyed-value pairs for `object`\nwhich can be consumed by `_.fromPairs`. If `object` is a map or set, its\nentries are returned.",
  "memberof": "_",
  "alias": "entries",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to query.",
      "name": "object"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.toPairs(new Foo);\n// => [['a', 1], ['b', 2]] (iteration order is not guaranteed)"
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Array"
        ]
      },
      "description": "Returns the key-value pairs."
    }
  ],
  "since": "4.0.0",
  "category": "Object",
  "meta": {
    "lineno": 28,
    "filename": "toPairs.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 162
}
    */
