// react-lodash/src/EntriesIn.js
import React from 'react'
import entriesIn from 'lodash/entriesIn'

export default function EntriesIn(props) {
  return props.children(entriesIn(props.object))
}

EntriesIn.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.entriesIn",
  "longname": "_.entriesIn",
  "name": "entriesIn",
  "kind": "member",
  "scope": "static",
  "description": "Creates an array of own and inherited enumerable string keyed-value pairs\nfor `object` which can be consumed by `_.fromPairs`. If `object` is a map\nor set, its entries are returned.",
  "memberof": "_",
  "alias": "entriesIn",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to query.",
      "name": "object"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.toPairsIn(new Foo);\n// => [['a', 1], ['b', 2], ['c', 3]] (iteration order is not guaranteed)"
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Array"
        ]
      },
      "description": "Returns the key-value pairs."
    }
  ],
  "since": "4.0.0",
  "category": "Object",
  "meta": {
    "lineno": 28,
    "filename": "toPairsIn.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 163
}
    */
