// react-lodash/src/ForInRight.js
import React from 'react'
import forInRight from 'lodash/forInRight'

export default function ForInRight(props) {
  return props.children(forInRight(props.object, props.iteratee))
}

ForInRight.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.forInRight",
  "longname": "_.forInRight",
  "name": "forInRight",
  "kind": "function",
  "scope": "static",
  "description": "This method is like `_.forIn` except that it iterates over properties of\n`object` in the opposite order.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to iterate over.",
      "name": "object"
    },
    {
      "type": {
        "names": [
          "function"
        ]
      },
      "optional": true,
      "defaultvalue": "_.identity",
      "description": "The function invoked per iteration.",
      "name": "iteratee"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.forInRight(new Foo, function(value, key) {\n  console.log(key);\n});\n// => Logs 'c', 'b', then 'a' assuming `_.forIn` logs 'a', 'b', then 'c'."
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "Returns `object`."
    }
  ],
  "since": "2.0.0",
  "category": "Object",
  "see": [
    "_.forIn"
  ],
  "meta": {
    "lineno": 31,
    "filename": "forInRight.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 230
}
    */
