// react-lodash/src/ForIn.js
import React from 'react'
import forIn from 'lodash/forIn'

export default function ForIn(props) {
  return props.children(forIn(props.object, props.iteratee))
}

ForIn.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.forIn",
  "longname": "_.forIn",
  "name": "forIn",
  "kind": "function",
  "scope": "static",
  "description": "Iterates over own and inherited enumerable string keyed properties of an\nobject and invokes `iteratee` for each property. The iteratee is invoked\nwith three arguments: (value, key, object). Iteratee functions may exit\niteration early by explicitly returning `false`.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to iterate over.",
      "name": "object"
    },
    {
      "type": {
        "names": [
          "function"
        ]
      },
      "optional": true,
      "defaultvalue": "_.identity",
      "description": "The function invoked per iteration.",
      "name": "iteratee"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.forIn(new Foo, function(value, key) {\n  console.log(key);\n});\n// => Logs 'a', 'b', then 'c' (iteration order is not guaranteed)."
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "Returns `object`."
    }
  ],
  "since": "0.3.0",
  "category": "Object",
  "see": [
    "_.forInRight"
  ],
  "meta": {
    "lineno": 33,
    "filename": "forIn.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 229
}
    */
