// react-lodash/src/ForOwn.js
import React from 'react'
import forOwn from 'lodash/forOwn'

export default function ForOwn(props) {
  return props.children(forOwn(props.object, props.iteratee))
}

ForOwn.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.forOwn",
  "longname": "_.forOwn",
  "name": "forOwn",
  "kind": "function",
  "scope": "static",
  "description": "Iterates over own enumerable string keyed properties of an object and\ninvokes `iteratee` for each property. The iteratee is invoked with three\narguments: (value, key, object). Iteratee functions may exit iteration\nearly by explicitly returning `false`.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to iterate over.",
      "name": "object"
    },
    {
      "type": {
        "names": [
          "function"
        ]
      },
      "optional": true,
      "defaultvalue": "_.identity",
      "description": "The function invoked per iteration.",
      "name": "iteratee"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.forOwn(new Foo, function(value, key) {\n  console.log(key);\n});\n// => Logs 'a' then 'b' (iteration order is not guaranteed)."
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "Returns `object`."
    }
  ],
  "since": "0.3.0",
  "category": "Object",
  "see": [
    "_.forOwnRight"
  ],
  "meta": {
    "lineno": 32,
    "filename": "forOwn.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 231
}
    */
