// react-lodash/src/ForOwnRight.js
import React from 'react'
import forOwnRight from 'lodash/forOwnRight'

export default function ForOwnRight(props) {
  return props.children(forOwnRight(props.object, props.iteratee))
}

ForOwnRight.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.forOwnRight",
  "longname": "_.forOwnRight",
  "name": "forOwnRight",
  "kind": "function",
  "scope": "static",
  "description": "This method is like `_.forOwn` except that it iterates over properties of\n`object` in the opposite order.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to iterate over.",
      "name": "object"
    },
    {
      "type": {
        "names": [
          "function"
        ]
      },
      "optional": true,
      "defaultvalue": "_.identity",
      "description": "The function invoked per iteration.",
      "name": "iteratee"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.forOwnRight(new Foo, function(value, key) {\n  console.log(key);\n});\n// => Logs 'b' then 'a' assuming `_.forOwn` logs 'a' then 'b'."
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "Returns `object`."
    }
  ],
  "since": "2.0.0",
  "category": "Object",
  "see": [
    "_.forOwn"
  ],
  "meta": {
    "lineno": 30,
    "filename": "forOwnRight.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 232
}
    */
