// react-lodash/src/Functions.js
import React from 'react'
import functions from 'lodash/functions'

export default function Functions(props) {
  return props.children(functions(props.object))
}

Functions.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.functions",
  "longname": "_.functions",
  "name": "functions",
  "kind": "function",
  "scope": "static",
  "description": "Creates an array of function property names from own enumerable properties\nof `object`.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to inspect.",
      "name": "object"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = _.constant('a');\n  this.b = _.constant('b');\n}\n\nFoo.prototype.c = _.constant('c');\n\n_.functions(new Foo);\n// => ['a', 'b']"
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Array"
        ]
      },
      "description": "Returns the function names."
    }
  ],
  "since": "0.1.0",
  "category": "Object",
  "see": [
    "_.functionsIn"
  ],
  "meta": {
    "lineno": 27,
    "filename": "functions.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 66
}
    */
