// react-lodash/src/KeysIn.js
import React from 'react'
import keysIn from 'lodash/keysIn'

export default function KeysIn(props) {
  return props.children(keysIn(props.object))
}

KeysIn.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.keysIn",
  "longname": "_.keysIn",
  "name": "keysIn",
  "kind": "function",
  "scope": "static",
  "description": "Creates an array of the own and inherited enumerable property names of `object`.\n\n**Note:** Non-object values are coerced to objects.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to query.",
      "name": "object"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.keysIn(new Foo);\n// => ['a', 'b', 'c'] (iteration order is not guaranteed)"
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Array"
        ]
      },
      "description": "Returns the array of property names."
    }
  ],
  "since": "3.0.0",
  "category": "Object",
  "meta": {
    "lineno": 28,
    "filename": "keysIn.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 87
}
    */
