// react-lodash/src/Values.js
import React from 'react'
import values from 'lodash/values'

export default function Values(props) {
  return props.children(values(props.object))
}

Values.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.values",
  "longname": "_.values",
  "name": "values",
  "kind": "function",
  "scope": "static",
  "description": "Creates an array of the own enumerable string keyed property values of `object`.\n\n**Note:** Non-object values are coerced to objects.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to query.",
      "name": "object"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.values(new Foo);\n// => [1, 2] (iteration order is not guaranteed)\n\n_.values('hi');\n// => ['h', 'i']"
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Array"
        ]
      },
      "description": "Returns the array of property values."
    }
  ],
  "since": "0.1.0",
  "category": "Object",
  "meta": {
    "lineno": 30,
    "filename": "values.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 179
}
    */
