// react-lodash/src/ValuesIn.js
import React from 'react'
import valuesIn from 'lodash/valuesIn'

export default function ValuesIn(props) {
  return props.children(valuesIn(props.object))
}

ValuesIn.defaultProps = {
  children: value => value
}
  
    /* Used for debugging
    {
  "id": "_.valuesIn",
  "longname": "_.valuesIn",
  "name": "valuesIn",
  "kind": "function",
  "scope": "static",
  "description": "Creates an array of the own and inherited enumerable string keyed property\nvalues of `object`.\n\n**Note:** Non-object values are coerced to objects.",
  "memberof": "_",
  "params": [
    {
      "type": {
        "names": [
          "Object"
        ]
      },
      "description": "The object to query.",
      "name": "object"
    }
  ],
  "examples": [
    "function Foo() {\n  this.a = 1;\n  this.b = 2;\n}\n\nFoo.prototype.c = 3;\n\n_.valuesIn(new Foo);\n// => [1, 2, 3] (iteration order is not guaranteed)"
  ],
  "returns": [
    {
      "type": {
        "names": [
          "Array"
        ]
      },
      "description": "Returns the array of property values."
    }
  ],
  "since": "3.0.0",
  "category": "Object",
  "meta": {
    "lineno": 28,
    "filename": "valuesIn.js",
    "path": "D:\\code\\react-lodash\\node_modules\\lodash"
  },
  "order": 180
}
    */
